### ISSUE TITLE
Refine Combo Component Keyboard Navigation & Escape Key Behavior

### ISSUE THREAD

**Issue by MonikaKirkova** [IgniteUI/igniteui-angular#16223]
## Is your feature request related to a problem? Please describe.
A previous change to keyboard navigation introduced behavior that disrupted the expected user flow when navigating the combo component. This needs to be reverted to restore intuitive navigation.
Reverting this change will remove the ability to clear selected values using keyboard navigation. Currently, there is no alternative way to clear the selection using only the keyboard, which impacts accessibility and keyboard-only workflows.

## Describe the solution you'd like
Revert the recent change in keyboard navigation to restore the original, more intuitive user experience.

Update Escape key behavior - Pressing Escape when the dropdown is closed should clear the selected value(s) in the combo component.

For the full specification please check the [specification](https://github.com/IgniteUI/igniteui-angular/wiki/IgxCombo-(Component)).
